pkg/providers/cache.go

```diff
@@ -9,19 +9,26 @@ import (
 	"github.com/open-sauced/pizza/oven/pkg/cache"
 )
 
+// Config provides the configuration set on server startup
+// - Never Evict Repos: Repos that are preserved in cache regardless of LRU policy
+type Config struct {
+	NeverEvictRepos map[string]int
+}
+
 // LRUCacheGitRepoProvider is a git repository provider that uses an internal
 // Least Recently Used cache on disk for loading and querying git repositories.
 // LRUCacheGitRepoProvider implements and statisfies the GitRepoProvider
 // interface.
 type LRUCacheGitRepoProvider struct {
 	logger   *zap.SugaredLogger
 	LRUCache *cache.GitRepoLRUCache
+	Config   *Config
 }
```

Comment:
A configuration specific to the LRU cache itself doesn't make too much sense in the context of a broader config being needed. I think you had it right with the server config but the `NeverEvict` slice is what needs to be passed to `NewLRUCacheGitRepoProvider`:

```go
func NewLRUCacheGitRepoProvider(cacheDir string, minFreeDisk uint64, l *zap.SugaredLogger, NeverEvictRepos &map[string]int) (GitRepoProvider, error)
```

and it'd get called something like:

```go
pizzaGitProvider, err = providers.NewLRUCacheGitRepoProvider(cacheDir, minFreeDiskUint64, sugarLogger, config.NeverEvict)
```

And it may make sense to denote `NeverEvictRepos` as its own type so we're not passing through a reference to a literal map as the type.

This way, the top level `Config` can latter be added onto without having to refactor abunch of configuration code on different providers.

Reply:
Thanks for the feedback! got it, should I keep the config passed to ```pizzaOvenSever``` ? for now I just defined the config struct there just in case for future configurations that might want to be passed there. I added the ```NeverEvictRepos ``` type and drilled that one to the LRU cache.